app/Main.hs

```diff
@@ -412,23 +412,33 @@ checkForUpdates applyMinorUpdates applyMajorUpdates = do
     isMinorReleaseFrom (x : xs) (y : ys) = y == x && ys > xs
     isMinorReleaseFrom _        _        = False
 
-verifyPackageSet :: IO ()
-verifyPackageSet = do
+verifyPackage :: Maybe String -> IO ()
+verifyPackage (Just name) = case mkPackageName (pack name) of
+  Right packageName -> verifyPackageSet (pure packageName)
+  Left pnError -> echoT . pack $ "Error while parsing input package name: " <> show pnError
+verifyPackage Nothing = echoT "A package name is required for verify"
+
+verifyPackageSet :: Maybe PackageName -> IO ()
+verifyPackageSet filterName = do
   pkg <- readPackageFile
   db <- readPackageSet pkg
+  let filteredDb = maybe id (flip Map.restrictKeys . Set.singleton) filterName $ db
 
-  echoT ("Verifying " <> pack (show (Map.size db)) <> " packages.")
-  echoT "Warning: this could take some time!"
+  case (Map.null filteredDb, filterName) of
+    (True, Just _) -> echoT . pack $ "No packages found with the name " <> show (runPackageName $ fromJust filterName)
```

Comment:
`fromJust` isn't needed here, the case on `filterName` already gives you the `PackageName`.